docs/user-guide/examples.md

```diff
@@ -130,4 +130,35 @@ defaultEntryPoints = ["http"]
     headerField = "X-WebAuth-User"
     [entryPoints.http.auth.basic]
     users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/", "test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]
+```
+
+
+## Enable authentication forwarding
+
+Uses a query param from the request to check for authentication with an auth service.
+If the auth service returns 200 OK the request is forwarded to the backend with additional headers/query params from the auth service.
+The response from the auth service is parsed, json encoded and can be forwarded to the backend as headers/query params.
+The field names in the request and the response can be manipulated for internal use via `as` config for request params 
+and the response from the auth server can be manipulated into a different name via `as` config for the response params.
+```
+defaultEntryPoints = ["http"]
+[entryPoints]
+  [entryPoints.http]
+  address = ":80"
+  [entryPoints.http.auth.forward]
+    address = "http://authserver.com/auth"
+    [entryPoints.http.auth.Forward.requestParameters.email]
+      name = "email"
+      as = "theEmail"
+    [entryPoints.http.auth.Forward.requestParameters.token]
+      name = "token"
+      as = "theToken"
+    [entryPoints.http.auth.Forward.responseReplayFields.userId]
+      path = "user.id"
+      as = "X-User-Id"
+      in = "header"
+    [entryPoints.http.auth.Forward.responseReplayFields.userName]
+      path = "user.name"
+      as = "" # No name transformation
+      in = "parameter"
```

Comment:
@Toflar I have implemented forwarding of headers, regarding "forwarding the full body" I agree with @diegooliveira it should be another PR I have started implementing it but there are some aspects that look strange for example:
1. I would need to force the request to be JSON but what will happen if the original request was XML will the auth server know how to handle it?
2. Sending random body (and I say random because there is no original route in the request) to the auth server will be without context regarding the destination of the request.
3. What HTTP method should be used since currently we use GET to send the query params to send the body we would need to post it and back to point 2 if we don't use the same verb or provide this context in the request we will have a problem and the body will be useless.
4. To overcome issues issues 2-3 we would have to do one of 2 things create some kind of generic message that will be posted to the auth server with route, method and body if such exists and implement in the auth server support for such a request or a less good solution is to forward the request to the auth server with the same method but loos the path along the way at which point your auth server would have to be able to get the query string params that we want to forward for authentication and read the body if one exists in addition support all http methods in one end point...

Those are the issues just from the first 5 minuets of the implementation. I really think it would be best to open a different PR for this.